src/main/java/com/abdullah/coding/challenge/services/AdminService.java

```diff
@@ -0,0 +1,67 @@
+package com.abdullah.coding.challenge.services;
+
+import com.abdullah.coding.challenge.Response;
+import com.abdullah.coding.challenge.dto.CabDTO;
+import com.abdullah.coding.challenge.entities.Booking;
+import com.abdullah.coding.challenge.entities.Cab;
+import com.abdullah.coding.challenge.enums.BookingStatus;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+
+@Service
+public class AdminService {
+    @Autowired
+    private BookingService bookingService;
+    @Autowired
+    private CabService cabService;
+    @Autowired
+    private CustomerService customerService;
+
+    public Response<?> cancelBooking(String bookingCode) {
+        return bookingService.cancelBooking(bookingCode);
+    }
+
+    @Transactional
```

Comment:
It denotes a transaction nature of a method because within that method, there are multiple repository operations that are dependent on each other. If any operation fails, it must discard any changes.